From 43667e60dc6e873e9f1cb7e68e6dc1ed8d376fe7 Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Tue, 21 Nov 2023 19:28:49 +0000
Subject: [PATCH 004/748] Revert "staging: vc04_services: bcm2835-audio: Drop
 MODULE_ALIAS"

This reverts commit 79ab0ca23abfc055f76a39839a89e03835bd06cc.
---
 drivers/staging/vc04_services/bcm2835-audio/bcm2835.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/staging/vc04_services/bcm2835-audio/bcm2835.c b/drivers/staging/vc04_services/bcm2835-audio/bcm2835.c
index b74cb104e9de..06bdc7673d4b 100644
--- a/drivers/staging/vc04_services/bcm2835-audio/bcm2835.c
+++ b/drivers/staging/vc04_services/bcm2835-audio/bcm2835.c
@@ -311,19 +311,12 @@ static int snd_bcm2835_alsa_resume(struct vchiq_device *device)
 
 #endif
 
-static struct vchiq_device_id device_id_table[] = {
-	{ .name = "bcm2835-audio" },
-	{}
-};
-MODULE_DEVICE_TABLE(vchiq, device_id_table);
-
 static struct vchiq_driver bcm2835_alsa_driver = {
 	.probe = snd_bcm2835_alsa_probe,
 #ifdef CONFIG_PM
 	.suspend = snd_bcm2835_alsa_suspend,
 	.resume = snd_bcm2835_alsa_resume,
 #endif
-	.id_table = device_id_table,
 	.driver = {
 		.name = "bcm2835-audio",
 	},
@@ -333,3 +326,4 @@ module_vchiq_driver(bcm2835_alsa_driver);
 MODULE_AUTHOR("Dom Cobley");
 MODULE_DESCRIPTION("Alsa driver for BCM2835 chip");
 MODULE_LICENSE("GPL");
+MODULE_ALIAS("bcm2835-audio");
-- 
2.43.0

